src/main/java/org/logstash/dissect/DelimiterFactory.java

```diff
@@ -0,0 +1,20 @@
+package org.logstash.dissect;
+
+import org.logstash.dissect.search.DoubleByteSearchStrategy;
+import org.logstash.dissect.search.MultiByteSearchStrategy;
+import org.logstash.dissect.search.SingleByteSearchStrategy;
+import org.logstash.dissect.search.ZeroByteSearchStrategy;
+
+class DelimiterFactory {
+    public static Delimiter create(String delim) {
```

Comment:
I think the factory only makes sense if you have more of them. If you have just one, I'd tend to implement `#create()` directly in `Delimiter`.